File: node/configure_model_ivr.js
const fs = require("fs");
const { SpeechClient } = require("@soniox/soniox-node");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
  const words = [];

  const onDataHandler = async (result) => {
    for (const word of result.words) {
        words.push(word);
        if (word.text == "<end>") {
            console.log(words.map((word) => word.text).join(" "));
            words.length = 0;
        }
    }
  };

  const onEndHandler = (error) => {
    if (error) {
        console.log("Error!", error);
    }
  };

  const stream = speechClient.transcribeStream(
    {
        model: "precision_ivr",
        enable_endpoint_detection: true,
        include_nonfinal: false,
    },
    onDataHandler,
    onEndHandler
  );

  const CHUNK_SIZE = 1024;
  const readable = fs.createReadStream("../test_data/test_audio_ivr.flac", {
    highWaterMark: CHUNK_SIZE,
  });

  for await (const chunk of readable) {
    await stream.writeAsync(chunk);
  }

  stream.end();
})();

File: python/configure_model_ivr.py
from typing import Iterable
from soniox.transcribe_live import transcribe_stream
from soniox.speech_service import SpeechClient


def iter_audio() -> Iterable[bytes]:
    with open("../test_data/test_audio_ivr.flac", "rb") as fh:
        while True:
            audio = fh.read(1024)
            if len(audio) == 0:
                break
            yield audio


def main():
    with SpeechClient() as client:
        words = []
        for result in transcribe_stream(
            iter_audio(),
            client,
            model="en_precision_ivr",
            enable_endpoint_detection=True,
            include_nonfinal=False,
        ):
            for word in result.words:
                words.append(word)
                if word.text == "<end>":
                    print(" ".join(w.text for w in words))
                    words = []


if __name__ == "__main__":
    main()
